realm/realm-library/src/main/java/io/realm/internal/SharedGroup.java

```diff
@@ -72,6 +81,51 @@ public SharedGroup(String canonicalPath, boolean enableImplicitTransactions, Dur
         checkNativePtrNotZero();
     }
 
+    private long openSharedGroupOrFail(Durability durability, byte[] key) {
+        // We have anecdotal evidence that on some versions of Android it is possible for two versions of an app
+        // to exist in two processes during an app upgrade. This is problematic since the lock file might not be
+        // compatible across two versions of Android. See https://github.com/realm/realm-java/issues/2459. If this
+        // happens we assume the overlap is really small so instead of failing outright we retry using incremental
+        // backoff.
+        int i = 0;
+        final long start = System.nanoTime();
+        RuntimeException lastError = null;
+        while (TimeUnit.MILLISECONDS.convert(System.nanoTime() - start, TimeUnit.NANOSECONDS) < INCREMENTAL_BACKOFF_LIMIT_MS) {
+            try {
+                long nativePtr = createNativeWithImplicitTransactions(nativeReplicationPtr, durability.value, key);
```

Comment:
I believe core overrides it if it is safe to do so, otherwise we should have had way more reports on this since it would happen for everybody when they did an upgrade.